pkgs/development/compilers/dmd/default.nix

```
@@ -1,12 +1,12 @@
 { stdenv, lib, fetchFromGitHub
-, makeWrapper, unzip, which, writeTextFile
-, curl, tzdata, gdb, Foundation, git, callPackage
+, makeWrapper, which, writeTextFile
+, curl, tzdata, gdb, Foundation, callPackage
 , targetPackages, fetchpatch, bash
 , HOST_DMD? "${callPackage ./bootstrap.nix { }}/bin/dmd"
-, version? "2.097.2"
-, dmdSha256? "16ldkk32y7ln82n7g2ym5d1xf3vly3i31hf8600cpvimf6yhr6kb"
-, druntimeSha256? "1sayg6ia85jln8g28vb4m124c27lgbkd6xzg9gblss8ardb8dsp1"
-, phobosSha256? "0czg13h65b6qwhk9ibya21z3iv3fpk3rsjr3zbcrpc2spqjknfw5"
+, version? "2.100.1"
+, dmdSha256? "sha256-14lCwD1jxDd1r29Wt/QQ0mvUAfCE0vhiJNlLIFeTz7Q="
+, druntimeSha256? "sha256-qXvY1ECN4mPwOGgOE1FWwvxoRvlSww3tGLWgBdhzAKo="
+, phobosSha256? "sha256-PcQdNfaoVyTXdDj65Xd2EMh6X2i/P9vOid5zAXThreM="
```

```suggestion
, version ? "2.100.1"
, dmdSha256 ? "sha256-14lCwD1jxDd1r29Wt/QQ0mvUAfCE0vhiJNlLIFeTz7Q="
, druntimeSha256 ? "sha256-qXvY1ECN4mPwOGgOE1FWwvxoRvlSww3tGLWgBdhzAKo="
, phobosSha256 ? "sha256-PcQdNfaoVyTXdDj65Xd2EMh6X2i/P9vOid5zAXThreM="
```
also we should consider inlining this

---

Whitespace corrections have been applied.

I agree that it's worth to consider for consistency, but I don't have much knowledge about the Dlang ecosystem. Pinging some of the other compiler maintainers for advice: @ThomasMader @dukc.

Both `dmd` and `ldc` are written to sort-of accommodate multiple packaged from-source versions, yet only 1 version of each is packaged in-tree. If this is a leftover from past times and the Dlang ecosystem should be fine with just 1 version of these then I think we should move this into the derivation body in both and simplify away all those version checks in `dmd`, if it's still required or there are other objections then I'd refactor this so `dmd` handles it like `ldc` does:

https://github.com/NixOS/nixpkgs/blob/e5ab352d0911d86e3625c3af2b54fc16c4c5cff2/pkgs/development/compilers/ldc/generic.nix#L1-L6
https://github.com/NixOS/nixpkgs/blob/e5ab352d0911d86e3625c3af2b54fc16c4c5cff2/pkgs/development/compilers/ldc/default.nix#L1-L4

---

I see, thanks for the explanation. I've made DMD similar to LDC w/r/t the generic/default split.

I've tested the oldest DMD version in your oldDDerivations repo and it additionally needs `git` in `nativeBuildInputs` and `unzip` in `checkInputs`, which 2.100.1 (and maybe some less-recent versions too) doesn't need anymore. Should I check when they stopped being requirements and add them as version-specific dependencies?

---

> Should I check when they stopped being requirements and add them as version-specific dependencies?

Better to just add them unconditionally, maybe with a "why" comment. Versioning things like this is something that works better in principle, but builds being slow it would always be poorly tested, so it's better to be conservative. Unless a NixPkgs "official" pops up with a contrary opinion.

---

Already gone ahead and did the testing, also expanded one of the test removals to an earlier version. :smile:

---

Well okay. I can remove them myself if they cause problems after all.